File: Java/binary search/DeepseekR1DistillLlama70b.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class DeepseekR1DistillLlama70b {

    /**
     * Searches for the specified target value in the given sorted array.
     * 
     * @param array  the sorted array to search in
     * @param target the value to search for
     * @return the index of the target value if found; otherwise, -1
     */
    public static int binarySearch(int[] array, int target) {
        return binarySearch(array, target, 0, array.length - 1);
    }

    /**
     * Helper method to perform the binary search within the specified range.
     * 
     * @param array  the sorted array to search in
     * @param target the value to search for
     * @param low    the starting index of the current search range
     * @param high   the ending index of the current search range
     * @return the index of the target value if found; otherwise, -1
     */
    private static int binarySearch(int[] array, int target, int low, int high) {
        while (low <= high) {
            int mid = low + (high - low) / 2; // Avoid integer overflow

            if (array[mid] == target) {
                return mid;
            } else if (array[mid] < target) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return -1;
    }

    // Example usage:
    public static void main(String[] args) {
        List<Integer> inputList = new ArrayList<>();

        // Ler o array ordenado de ../../inputs.txt
        try (BufferedReader br = new BufferedReader(new FileReader("../../inputs.txt"))) {
            String line;
            while ((line = br.readLine()) != null) {
                inputList.add(Integer.parseInt(line.trim()));
            }
        } catch (IOException e) {
            return;
        }

        int[] array = inputList.stream().mapToInt(Integer::intValue).toArray();

        int target = 15;
        Arrays.sort(array);
        int result = binarySearch(array, target);
    }
}

File: Python/binary search/deepseek-r1-distill-llama-70b.py
def binary_search(arr, target):
    """
    This function performs a binary search on a sorted array to find the index of the target value.
    
    Args:
        arr (list): A sorted list of elements.
        target (int/float): The value to be searched in the list.
    
    Returns:
        int: The index of the target value if found, otherwise -1.
    """
    low = 0
    high = len(arr) - 1
    
    while low <= high:
        mid = (low + high) // 2  # Using integer division to avoid floating point
        
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1  # Search in the right half
        else:
            high = mid - 1  # Search in the left half
    
    return -1  # Target not found in the array

import sys

def main():
    input_path = "inputs.txt" if len(sys.argv) < 2 else sys.argv[1]

    try:
        with open(input_path, 'r') as f:
            arr = [int(line.strip()) for line in f if line.strip()]
    except FileNotFoundError:
        return

    arr.sort()  # binary_search requires sorted array

    target = 15
    binary_search(arr, target)

if __name__ == "__main__":
    main()
